solr/solrj/src/java/org/apache/solr/client/solrj/impl/CloudHttp2SolrClient.java

```diff
@@ -422,7 +422,10 @@ public CloudHttp2SolrClient build() {
           }
         } else if (!solrUrls.isEmpty()) {
           try {
-            stateProvider = new Http2ClusterStateProvider(solrUrls, httpClient);
+            if (this.internalClientBuilder != null) {
+              this.httpClient = internalClientBuilder.build();
```

Comment:
I think there might be an issue here with the `httpClient` instance created here is not being closed correctly. not 100% sure I will need to check more closely. before this change I agree there was a default client being created lacking the correct params but it also closed.

Reply:
thanks for chiming in...  so, this change is making my TestExportTool blow up with unclosed httpclients..   I've banged my head on this for a LONG time, and not making headway.   What I can say is that the test with the BasicAuth demonstrates the problem, but I can't grok the soluiton...   @stillalex I would love your ideas if you wanted to take a stab at it..   it felt like such an easy fix!

Reply:
That's funny, I pushed 2 weeks ago an internal PR (at my work) for this exact same bug, as I faced it myself.

@stillalex  is correct, the resource won't be released if an exception is raised when instantiating the  _Http2ClusterStateProvider_

Plus we end up creating two _httpCient_ instances. CloudHttp2SolrClient and stateProvider would use 2 different _httClient_s.

In my PR, I pushed most of the code into the CloudHttp2SolrClient constructor. I added a test case to test if the same _httpClient_ is preserved in both _CloudHttp2SolrClient_ and _Http2ClusterStateProvider_.
All test cases passed, including _TestExportTool_.

Will see if I can push the PR here (upstream).

Reply:
Thanks all for looking at this.  I have definitly hit the "end" of my rope on this one, just can't figure it out...   My Java-fu isn't up to this one.   

I am inclined to close this PR in favour of what you can whip up..  However, since I really really really really really really want basic auth to work on all the tools, I am hesitating to actually do that!

Reply:
Actually I have 2 PRs that I am pretty sure would help a lot on this.
One that fixes this bug.
And anther one that I talked about [here](https://github.com/apache/solr/pull/1994#discussion_r1535512663) where I refactored ZK related SolrCli tools to separate _SolrZkClient_ and _CloudHttp2SolrClient_  instantiation/usage.

Will see if I can speed up the internal review and push them upstream.

Reply:
@epugh 
Here's the [PR](https://github.com/apache/solr/pull/2412)
Tested (locally) your testWithBasicAuth() code and the test passed